src/pages/Editor.vue

```diff
@@ -811,24 +829,42 @@ export default {
     },
     addNewItem() {
       this.$refs.playerObj.player.pause();
-      // get all the data needed to create a new item
       var newItem = {};
+
+      // get a valid timestamp for adding a new item
       var newTimestamp = this.getTimestampForNewItem();
       if (newTimestamp == -1) {
         this.showCannotAddItemDialog();
         return;
       }
-      // create the newItem object
-      newItem["time"] = newTimestamp;
-      newItem["type"] = this.getItemTypeForNewItem();
-      newItem["meta"] = this.getMetadataForNewItem();
-      newItem["details"] = this.getDetailsForNewItem();
 
-      // push it into items, update the itemTimestamps and currentItemIndex
-      this.items.push(newItem);
-      this.itemTimestamps = this.getItemTimestamps(this.items);
-      this.currentItemIndex = this.itemTimestamps.indexOf(this.currentTimestamp);
-      this.markItemSelected(this.currentItemIndex);
+      // create item, then create the question, then update local states
+      ItemService.createItem({
+        plio: this.plioDBId,
+        type: this.getItemTypeForNewItem(),
+        time: newTimestamp,
+        meta: this.getMetadataForNewItem(),
+      })
+        .then((createdItem) => {
+          newItem = createdItem;
```

Comment:
what is the use of this line?

Reply:
First we create an item, save the details into `newItem`, then we create a question and save the details into `newItem.details`.
The object that is being passed in the `then` handlers are item details and question details that have been created by the API calls. 
Combining both of those into one object, that is being done by using `newItem`
I can add comments to explain this more clearly. If a better name for the object comes to mind, do suggest

Reply:
Will explain on call